### PR TITLE
feat: Support latest identifier on checkins

### LINKED ISSUE #42816: Simplify monitor tracking API
### Problem Statement

Currently triggering a monitor check-in requires two calls:

-> POST .../monitors/{monitor_id}/checkins/

-> PUT .../monitors/{monitor_id}/checkins/{checkin_in}/ {"status": "ok"}

This creates a minor hurdle as it requires you to track the checkin_id returned in the first step, to publish the second step (to mark a job as finished, for example).

### Solution Brainstorm

Instead we could provide a simplified API along the lines of:

-> PUT .../monitors/{monitor_id}/checkins/latest/

This request would modify the most recent unfinished check-in, meaning a few behaviors could be implemented:

- With no checkins, this returns a 404 (same as invalid ID).
- With no unfinished checkin, this returns a 404 (same as invalid ID).
- With one unfinished checkin, this would modify that checkin.
- With multiple unfinished checkins, this would modify the most recent checkin. Meaning likely the previous checkin would time out and fail.

Refs GH-42283

### PR DESCRIPTION
**Pull request by dcramer**
Add the ability to target `latest` as the `checkin_id` parameter, which will pull the most recent (creation date) check-in that is not yet marked as finished.

Fixes GH-42816

### CODE REVIEW

File: src/sentry/api/base.py [hunk 1]
```diff
@@ -156,14 +163,14 @@ def build_cursor_link(self, request: Request, name, cursor):
             mutable_query_dict.pop("cursor")
             querystring = mutable_query_dict.urlencode()
 
-        base_url = absolute_uri(urlquote(request.path))
+        base_url = urlquote(request.build_absolute_uri())
```
Comment (dcramer) [inline src/sentry/api/base.py:166]: request is always present here so no need to use our internal abstraction

**Comment by dcramer** [comment]
I need to look at why we were using urlquote in the cursor as the link header seemingly needs to be an absolute uri (unquoted) based on running this new test on master

File: src/sentry/api/base.py [hunk 2]
```diff
@@ -156,14 +162,14 @@ def build_cursor_link(self, request: Request, name, cursor):
             mutable_query_dict.pop("cursor")
             querystring = mutable_query_dict.urlencode()
 
-        base_url = absolute_uri(urlquote(request.path))
+        base_url = request.build_absolute_uri()
```
Comment (dcramer) [inline src/sentry/api/base.py:165]: we have no tests for this case, so lets risk breakage here to uncover if theres a genuine problem. its possible our absolute_uri function does not handle the same conditions that build_absolute_uri does, but I cannot recall/come up with a scenario where we need to quote the path here.

the examples where this _might_ need to happen woudl be like if we had `/foo/bar/cool:story/`, and need to re-encode that to pass it as a new url

Comment (dcramer) [reply src/sentry/api/base.py]: retaining behavior as its cetainly expected to quote. added a test

**Review by davidenwang — APPROVED** [review]
new monitors code looks good to me

**Pull request merged**